### PR TITLE
Modify rule S1656: LaYC format

### PR DESCRIPTION
**Pull request by mary-georgiou-sonarsource**

### CODE REVIEW

File: rules/S1656/csharp/rule.adoc [hunk 1]
```diff
@@ -1,29 +1,45 @@
...
+[source,csharp,diff-id=1,diff-type=noncompliant]
 ----
-public void SetName(string name)
-{
-  name = name;
+class DataClass {
+  
+  static List<int> data;
+
+  public static void SetData(List<int> data)
```
Comment (martin-strecker-sonarsource) [inline rules/S1656/csharp/rule.adoc:19]: I liked the `this` qualifier example more than the static one.  The reasoning is that a mutable static field is bad coding practice (it behaves like a mutable global variable).

File: rules/S1656/csharp/rule.adoc [hunk 2]
```diff
@@ -1,29 +1,45 @@
 == Why is this an issue?
 
-include::../description.adoc[]
+Re-assigning a variable to itself is a defect as it has no actual effect and indicates meaning to do something else.
+It usually means that:
 
-=== Noncompliant code example
+* Either this statement is redundant and should be removed,
+* or the re-assignment is a mistake, and another value or variable was intended for the assignment instead
```
[the remark this answers is not in the document]
Comment (mary-georgiou-sonarsource) [reply rules/S1656/csharp/rule.adoc:8]: I'm not sure where. I added it to the resources. 
I'll change it if you mean something else.

File: rules/S1656/csharp/rule.adoc [hunk 3]
```diff
@@ -1,24 +1,40 @@
...
+=== Code examples
+
+==== Noncompliant code example
+
+[source,csharp,diff-id=1,diff-type=noncompliant]
 ----
-public void SetName(string name)
-{
-  name = name;
+class DataClass {
```
[the remark this answers is not in the document]
Comment (mary-georgiou-sonarsource) [reply rules/S1656/csharp/rule.adoc:15]: Great Thanks.
I went for the first one - it's simple and clear.

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**